terminator_test.go

```diff
@@ -277,6 +279,39 @@ func TestTerminator_Wait(t *testing.T) {
 	}
 }
 
+func TestTerminator_restores_signals(t *testing.T) {
```

Comment:
this test is more like a `integration` one, introducing a dedicated env variable to simulate the behavior.
To test the actual library end-to-end a separate integration test suite can be added, that can spawn a simple app with graterm configured and webserver waiting for requests


This timing-specific test relying on the scheduler are tricky to implement right, consider [mocking time](https://github.com/benbjohnson/clock) adding assertions that `Sleep` method was called

Reply:
This test actually is not timing-specific because it tests signal default handler restoration. I'll try to test without sleeps between signals.

Reply:
it's scheduler timing-specific because you rely on `time.Sleep` which is not a good practice, tests are likely to be flaky especially on lower CPU resources machines.

Reply:
I removed sleeps before sending SIGINT's locally. Test doesn't fail. So it shouldn't be flaky at all because we're waiting for process termination and checking for specific flags.

Reply:
@xakep666 now it looks better, thank you. What about having a separate integration test suite in a separate file ? So we could leverage integration testing with this approach for other cases as well and not mix it with unit tests

Reply:
It's good idea but `Suite` from testify doesn't allow parallel runs so maybe just move this test to separate file. Probably with specific tag.